core/api/api.hpp

```diff
@@ -154,6 +161,28 @@ namespace fc::api {
     API_METHOD(SyncSubmitBlock, void, const BlockMsg &)
 
     API_METHOD(WalletSign, Signature, const Address &, const Buffer &)
+
+    /**
+     * Lists the providers in the storage market state
+     * @return vector of StorageProviderInfo
+     */
+    API_METHOD(StateListStorageProviders, std::vector<StorageProviderInfo>)
+
+    /**
+     * List client deals
+     */
+    API_METHOD(StateListClientDeals, std::vector<StorageDeal>, const Address &)
+
+    /**
+     * Adds funds with the StorageMinerActor for a storage participant. Used by
+     * both providers and clients.
+     */
+    API_METHOD(AddFunds, void, const Address &, const TokenAmount &)
+    API_METHOD(ValidateAskSignature,
+               bool,
+               const SignedStorageAsk &,
+               const TipsetKey &)
+    API_METHOD(SignProposal, ClientDealProposal, Address, DealProposal)
```

Comment:
```suggestion
    API_METHOD(StateAccountKey, Address, const Address &, const TipsetKey &)
```
Then sign using keystore